reflex/app.py

```diff
@@ -659,81 +654,119 @@ def compile(self):
-        compile_results.append(compiler.compile_app(app_root))
+            # Compile the app wrapper.
+            app_root = self._app_root(app_wrappers=app_wrappers)
+            submit_work(compiler.compile_app, app_root)
 
-        # Get the compiled pages.
-        compile_results.extend(result.get() for result in page_futures)
+            # Compile the custom components.
+            submit_work(compiler.compile_components, custom_components)
 
-        # TODO the compile tasks below may also benefit from parallelization too
+            # Compile the root stylesheet with base styles.
+            submit_work(compiler.compile_root_stylesheet, self.stylesheets)
 
-        # Compile the custom components.
-        compile_results.append(compiler.compile_components(custom_components))
+            # Compile the root document.
+            submit_work(compiler.compile_document_root, self.head_components)
 
-        # Iterate through all the custom components and add their imports to the all_imports
-        for component in custom_components:
-            all_imports.update(component.get_imports())
+            # Compile the theme.
+            submit_work(compiler.compile_theme, style=self.style)
 
-        # Compile the root stylesheet with base styles.
-        compile_results.append(compiler.compile_root_stylesheet(self.stylesheets))
+            # Compile the contexts.
+            submit_work(compiler.compile_contexts, self.state)
 
-        # Compile the root document.
-        compile_results.append(compiler.compile_document_root(self.head_components))
+            # Compile the Tailwind config.
+            if config.tailwind is not None:
+                config.tailwind["content"] = config.tailwind.get(
+                    "content", constants.Tailwind.CONTENT
+                )
+            submit_work(compiler.compile_tailwind, config.tailwind)
 
-        # Compile the theme.
-        compile_results.append(compiler.compile_theme(style=self.style))
+            # Get imports from AppWrap components.
+            all_imports.update(app_root.get_imports())
 
-        # Compile the contexts.
-        compile_results.append(compiler.compile_contexts(self.state))
+            # Iterate through all the custom components and add their imports to the all_imports.
+            for component in custom_components:
+                all_imports.update(component.get_imports())
 
-        # Compile the Tailwind config.
-        if config.tailwind is not None:
-            config.tailwind["content"] = config.tailwind.get(
-                "content", constants.Tailwind.CONTENT
-            )
-            compile_results.append(compiler.compile_tailwind(config.tailwind))
+            # Empty the .web pages directory.
+            compiler.purge_web_pages_dir()
 
-        # Empty the .web pages directory
-        compiler.purge_web_pages_dir()
+            # Install frontend packages.
+            self.get_frontend_packages(all_imports)
```

Comment:
Not sure if it matters, but the compile bar flickers when run in progress with this command for me